content/en/docs/tutorials/configuration/updating-configuration-via-a-configmap.md

```diff
@@ -0,0 +1,675 @@
+
+{{% code_sample file="deployments/deployment-with-configmap-and-sidecar-container.yaml" %}}
+
+Create the Deployment:
+```shell
+kubectl apply -f https://k8s.io/examples/deployments/deployment-with-configmap-and-sidecar-container.yaml
+```
+
+Check the pods for this Deployment to ensure they are ready (matching by
+{{< glossary_tooltip text="selector" term_id="selector" >}}):
+```shell
+kubectl get pods --selector=app.kubernetes.io/name=configmap-sidecar-container
+```
+You should see an output similar to:
+```
+NAME                                           READY   STATUS    RESTARTS   AGE
+configmap-sidecar-container-5fb59f558b-87rp7   2/2     Running   0          94s
+configmap-sidecar-container-5fb59f558b-ccs7s   2/2     Running   0          94s
+configmap-sidecar-container-5fb59f558b-wnmgk   2/2     Running   0          94s
+````
+
+Expose the Deployment (the `kubectl` tool creates a
+{{<glossary_tooltip text="Service" term_id="service">}}  for you):
+```shell
+kubectl expose deployment configmap-sidecar-container --name=configmap-sidecar-service --port=8081 --target-port=80
+```
+
+Use ```kubectl``` to forward the port:
+```shell
+kubectl port-forward service/configmap-sidecar-service 8081:8081 & # this stays running in the background
+```
+
+Access the service.
+```shell
+curl http://localhost:8081
+```
+
+You should see an output similar to:
+```
+Sat Feb 17 13:09:05 UTC 2024 My preferred color is blue
+```
+
+Edit the ConfigMap:
+```shell
+kubectl edit configmap color
+```
+
+In the editor that appears, change the value of key `color` from `blue` to `green`. Save your changes.
+The kubectl tool updates the ConfigMap accordingly (if you see an error, try again).
+
+Here's an example of how that manifest could look after you edit it:
+
+```yaml
+apiVersion: v1
+data:
+  color: green
+kind: ConfigMap
+# You can leave the existing metadata as they are.
+# The values you'll see won't exactly match these.
+metadata:
+  creationTimestamp: "2024-02-17T12:20:30Z"
+  name: color
+  namespace: default
+  resourceVersion: "1054"
+  uid: e40bb34c-58df-4280-8bea-6ed16edccfaa
+```
+
+Loop over the service URL for few seconds.
+```shell
+# Cancel this when you're happy with it (Ctrl-C)
+while true; do curl --connect-timeout 7.5 http://localhost:8081; sleep 10; done
+```
+
+You should see the output change as follows:
+```
+Sat Feb 17 13:12:35 UTC 2024 My preferred color is blue
+Sat Feb 17 13:12:45 UTC 2024 My preferred color is blue
+Sat Feb 17 13:12:55 UTC 2024 My preferred color is blue
+Sat Feb 17 13:13:05 UTC 2024 My preferred color is blue
+Sat Feb 17 13:13:15 UTC 2024 My preferred color is green
+Sat Feb 17 13:13:25 UTC 2024 My preferred color is green
+Sat Feb 17 13:13:35 UTC 2024 My preferred color is green
+```
+
+## Update configuration via an immutable ConfigMap that is mounted as a Volume {#rollout-configmap-immutable-volume}
```

Comment:
nit: I wouldn't capitalize Volume as it is not an API kind.

Reply:
Ack. This will be handled in the next commit.